syntax = "proto3";

package travel_base;

message Coordinate{
double lat = 1;
double lng = 2;
}

message Stop{
string name = 1;
Coordinate coordinate = 2;
uint64 id = 3;
}

message Bus{
string name =1;
repeated uint64 id_stop = 2; // vector stops;
bool circle = 3;
}

message Length{
uint64 stop_from_id = 1;
uint64 stop_to_id = 2;
double length = 3;
}

import "map_renderer.proto";
import "transport_router.proto";
import "graph.proto";

message DB{
repeated Stop stops =1;
repeated Bus buses = 2;
repeated Length lengths= 3;
RenderSetting render_setting = 4;
RoutingSettings routing_settings = 5;
Graph graph = 6;
}
